bodhi/models/models.py

```diff
@@ -1448,12 +1458,42 @@ def unpush(self):
         self.untag()
 
         for build in self.builds:
-            koji.tagBuild(self.candidate_tag, build.nvr, force=True)
+            koji.tagBuild(self.release.candidate_tag, build.nvr, force=True)
 
         self.pushed = False
         self.status = UpdateStatus.unpushed
         mail.send_admin('unpushed', self)
 
+    def revoke(self):
+        """ Remove pending request for this update """
+        log.debug("Revoking %s" % self.title)
+
+        if self.status is not UpdateStatus.pending:
+            raise BodhiException("Can only revoke a pending update, not "
+                                 "one that is %s" % self.status.description)
+
+        # Remove the 'pending' koji tags from this update so taskotron stops
+        # evalulating them.
+        if self.request is UpdateRequest.testing:
+            self.remove_tag(self.release.pending_testing_tag)
+        elif self.request is UpdateRequest.stable:
+            self.remove_tag(self.release.pending_stable_tag)
+
+        # If we're revoking a request we need to set the status back to what it
+        # was before the request.  The only way to tell what status this update
+        # was in before the request is to look at koji tags.
+        current_tags = self.get_tags()
+        if self.release.testing_tag in current_tags:
+            self.status = UpdateRequest.testing
+        elif self.release.stable_tag in current_tags:
+            self.status = UpdateRequest.stable
+        else:
+            self.status = None
+
+        self.request = None
+
+        mail.send_admin('revoked', self)
```

Comment:
:+1:  I was just keeping the pattern in place that was already there (obsolete sent a mail, so revoke should too).